src/coin_test/data/dataset.py

```diff
@@ -0,0 +1,102 @@
+"""Define the Dataset class."""
+
+from collections import Counter
+
+import pandas as pd
+
+from .processors import Processor
+
+
+REQUIRED_COLS = {
+    "Open Time": int,
+    "Open": float,
+    "High": float,
+    "Low": float,
+    "Close": float,
+    "Volume": float,
+    "Close Time": int,
+}
+
+
+class Dataset:
+    """Manages a dataset."""
+
+    def __init__(
+        self,
+        asset: str,
```

Comment:
Discussion about metadata object, wrapping currency, asset, and  interval. Eamon may add this in a later API PR.